doc: Adds 1.38.0 upgrade guide with new features and bug fixes (#3489)
mongodb/terraform-provider-mongodbatlas: docs/guides/1.38.0-upgrade-guide.md

---
page_title: "Upgrade Guide 1.38.0"
---

> # MongoDB Atlas Provider 1.38.0: Upgrade and Information Guide

Reviewer: Should we add the general note at the bottom to flag other things mentioned in https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CHANGELOG.md that are not added here? (eg the deprecated unused fields)

Author: What about this sentence at the bottom:

The 1.38.0 release also includes general improvements, bug fixes and several key documentation updates. See the [CHANGELOG](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CHANGELOG.md) for more specific information.

Reviewer: yup I think that's the usual sentence we add

The Terraform MongoDB Atlas Provider version 1.38.0 introduces the following new features and fixes:

## New Resources, Data Sources, and Features

> - You can now use `mongodbatlas_api_key_project_assignment` to assign an api key to a project. See the [Migration Guide: Project API Key to API Key + Project Assignment](project-api-key-migration) to learn more.

Reviewer 2: i would link to the resource doc and the migration guide, not only the migration guide

Author: Thank you. Added in ff1e06f2088b6e2df90a1044e1db24798d05fec8

Author: Thank you. Added in ff1e06f2088b6e2df90a1044e1db24798d05fec8

> - You can now use `mongodbatlas_api_key_project_assignment` to assign an api key to a project. See the [Migration Guide: Project API Key to API Key + Project Assignment](project-api-key-migration) and the [resource documentation](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/api_key_project_assignment) to learn more.

Reviewer 3: ```suggestion
- You can now use `mongodbatlas_api_key_project_assignment` to assign an API key to a project. See the [Migration Guide: Project API Key to API Key + Project Assignment](project-api-key-migration) and the [resource documentation](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/api_key_project_assignment) to learn more.
```

> - The `mongodbatlas_organization` resource now supports import. See the [import guide](importing-organization) to learn more.

Reviewer 3: [nit] Suggest being a little more descriptive here (will defer to you for exact wording)
```suggestion
- The `mongodbatlas_organization` resource now supports importing an existing organization. See the [import guide](importing-organization) to learn more.
```

- The `mongodbatlas_third_party_integration` resource now supports the `send_user_provided_resource_tags` attribute to support sending `$querystats` to Datadog. See the [resource documentation](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/third_party_integration#DATADOG-1) to learn more.

## Bug Fixes

> - A pagination fix for plural data sources were only a single item was returned by default when `items_per_page` was undefined.

Reviewer 3: ```suggestion
- Fixes an issue with pagination for plural data sources where only a single item was returned by default when `items_per_page` was undefined.
```
